src/layers/upsample.jl

```diff
@@ -0,0 +1,81 @@
+"""
+  Upsample(mode = :nearest; scale = nothing, size = nothing)  
+
+An upsampling layer. 
+
+`scale` is a number or a tuple of numbers 
+representing  the output rescaling factor along each spatial dimension.
+For integer `scale`, all but the last 2 dimensions (channel and batch)
+will be rescaled by the same factor. 
+
+It is also possible to directly specify the output spatial `size`,
+as an alternative to using `scale`.
```

Comment:
The `nothing`s here are an implementation detail, 
```suggestion
  Upsample(mode = :nearest; [scale, size])  

An upsampling layer. One of two keywords must be given:

If `scale` is a number, this applies to all but the last two dimensions (channel and batch) of the input. 
It may also be a tuple, to control dimensions individually. Alternatively, keyword 
`size` accepts a tuple, to directly specify the leading dimensions of the output.
```

Reply:
Maybe we should also link to what the upsampling layer does.

Reply:
You mean to the NNlib functions? Those are already linked below.

Reply:
No, I mean describing what upsampling is. The nnlib functions are implementations of specific interpolation routines.

Link to super resolution gan or reducing the memory bandwidth of a network.